test flake: TestPGCoordinatorDual_Mainline

Seen here: https://github.com/coder/coder/actions/runs/6344009627/job/17233161856?pr=9920

<details>
<summary>Output:</summary>

```
=== FAIL: enterprise/tailnet TestPGCoordinatorDual_Mainline (60.04s)
    t.go:84: 2023-09-28 20:40:13.479 [debu]  coord1.pgcoord.querier: subscribed to client updates  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:13.482 [debu]  coord1.pgcoord.querier: subscribed to agent updates  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:13.482 [info]  coord1.pgcoord: starting coordinator  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:13.482 [debu]  coord2.pgcoord.querier: subscribed to client updates  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880
    t.go:84: 2023-09-28 20:40:13.482 [debu]  coord2.pgcoord.querier: subscribed to agent updates  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880
    t.go:84: 2023-09-28 20:40:13.482 [info]  coord2.pgcoord: starting coordinator  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880
    t.go:84: 2023-09-28 20:40:13.482 [info]  coord2.pgcoord: serving connection  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880
    t.go:84: 2023-09-28 20:40:13.482 [info]  coord1.pgcoord: serving connection  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  name=agent1
    t.go:84: 2023-09-28 20:40:13.482 [info]  coord2.pgcoord: serving connection  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  name=agent2
    t.go:84: 2023-09-28 20:40:13.482 [info]  coord1.pgcoord: serving connection  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:13.482 [info]  coord1.pgcoord: serving connection  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:13.489 [debu]  coord2.pgcoord: cleaned up old coordinators  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880
    t.go:84: 2023-09-28 20:40:13.490 [debu]  coord1.pgcoord: cleaned up old coordinators  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:13.491 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:13.493 [info]  coord2.pgcoord: serving connection  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880
    t.go:84: 2023-09-28 20:40:13.493 [debu]  coord1.pgcoord: got node update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  node={"id":0,"as_of":"0001-01-01T00:00:00Z","key":"nodekey:0000000000000000000000000000000000000000000000000000000000000000","disco":"discokey:0000000000000000000000000000000000000000000000000000000000000000","preferred_derp":11,"derp_latency":null,"derp_forced_websockets":null,"addresses":null,"allowed_ips":null,"endpoints":null}
    t.go:84: 2023-09-28 20:40:13.494 [debu]  coord2.pgcoord: got heartbeat  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  other_coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:13.494 [info]  coord2.pgcoord: heartbeats (re)started  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  other_coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:13.494 [debu]  coord2.pgcoord: set initial heartbeat timeout  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880
    t.go:84: 2023-09-28 20:40:13.494 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:13.494 [debu]  coord1.pgcoord.querier: skipping 0 length node update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  kind=2
    t.go:84: 2023-09-28 20:40:13.494 [debu]  coord1.pgcoord.querier: skipping 0 length node update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  kind=1
    t.go:84: 2023-09-28 20:40:13.494 [debu]  coord1.pgcoord.querier: skipping 0 length node update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  kind=1
    t.go:84: 2023-09-28 20:40:13.498 [debu]  coord1.pgcoord.querier: queried clients of agent  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  num_clients=0  error=<nil>
    t.go:84: 2023-09-28 20:40:13.498 [debu]  coord1.pgcoord.querier: sending mappings  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  mapping_len=0
    t.go:84: 2023-09-28 20:40:13.498 [debu]  coord1.pgcoord.querier: skipping 0 length node update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  kind=2
    t.go:84: 2023-09-28 20:40:13.502 [debu]  coord1.pgcoord: upserted client binding  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=801ff607-3a63-404a-9c7d-932ec133e23d  node="{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":11,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}"  error=<nil>
    t.go:84: 2023-09-28 20:40:13.504 [debu]  coord2.pgcoord: sent heartbeat  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880
    t.go:84: 2023-09-28 20:40:13.504 [debu]  coord2.pgcoord.querier: skipping 0 length node update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  kind=2
    t.go:84: 2023-09-28 20:40:13.504 [debu]  coord2.pgcoord: ignoring our own heartbeat  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880
    t.go:84: 2023-09-28 20:40:13.504 [debu]  coord2.pgcoord.querier: skipping 0 length node update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  kind=1
    t.go:84: 2023-09-28 20:40:13.504 [debu]  coord1.pgcoord: got heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  other_coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880
    t.go:84: 2023-09-28 20:40:13.504 [info]  coord1.pgcoord: heartbeats (re)started  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  other_coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880
    t.go:84: 2023-09-28 20:40:13.504 [debu]  coord1.pgcoord: set initial heartbeat timeout  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:13.504 [debu]  coord1.pgcoord.querier: skipping 0 length node update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f[339](https://github.com/coder/coder/actions/runs/6344009627/job/17233161856?pr=9920#step:5:340)  kind=1
    t.go:84: 2023-09-28 20:40:13.504 [debu]  coord1.pgcoord.querier: skipping 0 length node update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  kind=2
    t.go:84: 2023-09-28 20:40:13.504 [debu]  coord1.pgcoord.querier: skipping 0 length node update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  kind=1
    t.go:84: 2023-09-28 20:40:13.504 [debu]  coord2.pgcoord.querier: skipping 0 length node update  coordinator_id=[343](https://github.com/coder/coder/actions/runs/6344009627/job/17233161856?pr=9920#step:5:344)5eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  kind=1
    t.go:84: 2023-09-28 20:40:13.505 [debu]  coord2.pgcoord.querier: queried clients of agent  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  num_clients=0  error=<nil>
    t.go:84: 2023-09-28 20:40:13.505 [debu]  coord2.pgcoord.querier: sending mappings  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  mapping_len=0
    t.go:84: 2023-09-28 20:40:13.505 [debu]  coord2.pgcoord.querier: skipping 0 length node update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  kind=2
    t.go:84: 2023-09-28 20:40:13.510 [debu]  coord2.pgcoord.querier: got client update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  client_id=0ef4df4a-393a-4655-b98d-da4d1b93f6e6  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339
    t.go:84: 2023-09-28 20:40:13.510 [debu]  coord2.pgcoord.querier: ignoring update because we have no mapper  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  client_id=0ef4df4a-393a-4655-b98d-da4d1b93f6e6  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339
    t.go:84: 2023-09-28 20:40:13.510 [debu]  coord1.pgcoord.querier: got client update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=0ef4df4a-393a-4655-b98d-da4d1b93f6e6  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339
    t.go:84: 2023-09-28 20:40:13.512 [debu]  coord2.pgcoord: upserted client subscription  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  client_id=0ef4df4a-393a-4655-b98d-da4d1b93f6e6  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  error=<nil>
    t.go:84: 2023-09-28 20:40:13.512 [debu]  coord1.pgcoord.querier: queried agents  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  num_agents=0  error=<nil>
    t.go:84: 2023-09-28 20:40:13.512 [debu]  coord1.pgcoord.querier: sending mappings  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  mapping_len=0
    t.go:84: 2023-09-28 20:40:13.512 [debu]  coord1.pgcoord.querier: skipping 0 length node update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  kind=1
    t.go:84: 2023-09-28 20:40:13.514 [debu]  coord1.pgcoord.querier: queried clients of agent  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  num_clients=0  error=<nil>
    t.go:84: 2023-09-28 20:40:13.514 [debu]  coord1.pgcoord.querier: sending mappings  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  mapping_len=0
    t.go:84: 2023-09-28 20:40:13.514 [debu]  coord1.pgcoord.querier: skipping 0 length node update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  kind=2
    t.go:84: 2023-09-28 20:40:13.515 [debu]  coord2.pgcoord.querier: got client update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  client_id=edd376b1-729a-4fdc-8efa-bb8c4fa04452  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a
    t.go:84: 2023-09-28 20:40:13.515 [debu]  coord1.pgcoord.querier: got client update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=edd376b1-729a-4fdc-8efa-bb8c4fa04452  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a
    t.go:84: 2023-09-28 20:40:13.515 [debu]  coord1.pgcoord.querier: ignoring update because we have no mapper  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=edd376b1-729a-4fdc-8efa-bb8c4fa04452  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a
    t.go:84: 2023-09-28 20:40:13.515 [debu]  coord2.pgcoord: upserted client subscription  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  client_id=edd376b1-729a-4fdc-8efa-bb8c4fa04452  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  error=<nil>
    t.go:84: 2023-09-28 20:40:13.515 [debu]  coord1.pgcoord: upserted client subscription  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=801ff607-3a63-404a-9c7d-932ec133e23d  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  error=<nil>
    t.go:84: 2023-09-28 20:40:13.515 [debu]  coord2.pgcoord.querier: got client update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  client_id=801ff607-3a63-404a-9c7d-932ec133e23d  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339
    t.go:84: 2023-09-28 20:40:13.515 [debu]  coord2.pgcoord.querier: ignoring update because we have no mapper  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  client_id=801ff607-3a63-404a-9c7d-932ec133e23d  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339
    t.go:84: 2023-09-28 20:40:13.515 [debu]  coord2.pgcoord.querier: queried agents  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  num_agents=0  error=<nil>
    t.go:84: 2023-09-28 20:40:13.515 [debu]  coord2.pgcoord.querier: sending mappings  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  mapping_len=0
    t.go:84: 2023-09-28 20:40:13.515 [debu]  coord1.pgcoord.querier: got client update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=801ff607-3a63-404a-9c7d-932ec133e23d  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339
    t.go:84: 2023-09-28 20:40:13.515 [debu]  coord2.pgcoord.querier: skipping 0 length node update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  kind=1
    t.go:84: 2023-09-28 20:40:13.517 [debu]  coord2.pgcoord.querier: queried clients of agent  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  num_clients=0  error=<nil>
    t.go:84: 2023-09-28 20:40:13.517 [debu]  coord2.pgcoord.querier: sending mappings  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  mapping_len=0
    t.go:84: 2023-09-28 20:40:13.517 [debu]  coord2.pgcoord.querier: skipping 0 length node update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  kind=2
    t.go:84: 2023-09-28 20:40:13.518 [debu]  coord1.pgcoord.querier: queried clients of agent  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  num_clients=1  error=<nil>
    t.go:84: 2023-09-28 20:40:13.518 [debu]  coord1.pgcoord.querier: sending mappings  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  mapping_len=1
    t.go:84: 2023-09-28 20:40:13.519 [debu]  coord1.pgcoord: wrote nodes  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  name=agent1  nodes="[{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":11,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}]"
    t.go:84: 2023-09-28 20:40:13.520 [debu]  coord2.pgcoord: got node update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  node={"id":0,"as_of":"0001-01-01T00:00:00Z","key":"nodekey:0000000000000000000000000000000000000000000000000000000000000000","disco":"discokey:0000000000000000000000000000000000000000000000000000000000000000","preferred_derp":21,"derp_latency":null,"derp_forced_websockets":null,"addresses":null,"allowed_ips":null,"endpoints":null}
    t.go:84: 2023-09-28 20:40:13.520 [debu]  coord1.pgcoord.querier: queried agents  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  num_agents=0  error=<nil>
    t.go:84: 2023-09-28 20:40:13.520 [debu]  coord1.pgcoord.querier: sending mappings  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  mapping_len=0
    t.go:84: 2023-09-28 20:40:13.520 [debu]  coord1.pgcoord.querier: skipping 0 length node update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  kind=1
    t.go:84: 2023-09-28 20:40:13.524 [debu]  coord2.pgcoord.querier: got client update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  client_id=0ef4df4a-393a-4655-b98d-da4d1b93f6e6  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339
    t.go:84: 2023-09-28 20:40:13.524 [debu]  coord2.pgcoord.querier: ignoring update because we have no mapper  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  client_id=0ef4df4a-393a-4655-b98d-da4d1b93f6e6  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339
    t.go:84: 2023-09-28 20:40:13.524 [debu]  coord1.pgcoord.querier: got client update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=0ef4df4a-393a-4655-b98d-da4d1b93f6e6  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339
    t.go:84: 2023-09-28 20:40:13.524 [debu]  coord2.pgcoord: upserted client binding  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  client_id=0ef4df4a-393a-4655-b98d-da4d1b93f6e6  node="{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":21,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}"  error=<nil>
    t.go:84: 2023-09-28 20:40:13.524 [debu]  coord2.pgcoord.querier: queried agents  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  num_agents=0  error=<nil>
    t.go:84: 2023-09-28 20:40:13.524 [debu]  coord2.pgcoord.querier: sending mappings  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  mapping_len=0
    t.go:84: 2023-09-28 20:40:13.524 [debu]  coord2.pgcoord.querier: skipping 0 length node update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  kind=1
    t.go:84: 2023-09-28 20:40:13.525 [debu]  coord1.pgcoord.querier: queried clients of agent  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  num_clients=2  error=<nil>
    t.go:84: 2023-09-28 20:40:13.525 [debu]  coord1.pgcoord.querier: sending mappings  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  mapping_len=2
    t.go:84: 2023-09-28 20:40:13.525 [debu]  coord2.pgcoord: got node update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  node={"id":0,"as_of":"0001-01-01T00:00:00Z","key":"nodekey:0000000000000000000000000000000000000000000000000000000000000000","disco":"discokey:0000000000000000000000000000000000000000000000000000000000000000","preferred_derp":22,"derp_latency":null,"derp_forced_websockets":null,"addresses":null,"allowed_ips":null,"endpoints":null}
    t.go:84: 2023-09-28 20:40:13.525 [debu]  coord1.pgcoord: wrote nodes  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  name=agent1  nodes="[{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":11,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null},{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":21,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}]"
    t.go:84: 2023-09-28 20:40:13.531 [debu]  coord2.pgcoord: upserted client binding  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  client_id=edd376b1-729a-4fdc-8efa-bb8c4fa04452  node="{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":22,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}"  error=<nil>
    t.go:84: 2023-09-28 20:40:13.531 [debu]  coord2.pgcoord.querier: got client update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  client_id=edd376b1-729a-4fdc-8efa-bb8c4fa04452  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a
    t.go:84: 2023-09-28 20:40:13.531 [debu]  coord1.pgcoord.querier: got client update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=edd376b1-729a-4fdc-8efa-bb8c4fa04452  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a
    t.go:84: 2023-09-28 20:40:13.531 [debu]  coord1.pgcoord.querier: ignoring update because we have no mapper  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=edd376b1-729a-4fdc-8efa-bb8c4fa04452  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a
    t.go:84: 2023-09-28 20:40:13.531 [debu]  coord2.pgcoord.querier: queried clients of agent  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  num_clients=1  error=<nil>
    t.go:84: 2023-09-28 20:40:13.531 [debu]  coord2.pgcoord.querier: sending mappings  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  mapping_len=1
    t.go:84: 2023-09-28 20:40:13.531 [debu]  coord2.pgcoord: got node update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  name=agent2  node={"id":0,"as_of":"0001-01-01T00:00:00Z","key":"nodekey:0000000000000000000000000000000000000000000000000000000000000000","disco":"discokey:0000000000000000000000000000000000000000000000000000000000000000","preferred_derp":2,"derp_latency":null,"derp_forced_websockets":null,"addresses":null,"allowed_ips":null,"endpoints":null}
    t.go:84: 2023-09-28 20:40:13.531 [debu]  coord2.pgcoord: wrote nodes  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  name=agent2  nodes="[{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":22,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}]"
    t.go:84: 2023-09-28 20:40:13.532 [debu]  coord2.pgcoord.querier: got agent update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a
    t.go:84: 2023-09-28 20:40:13.532 [debu]  coord2.pgcoord: upserted agent binding  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  node="{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":2,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}"  error=<nil>
    t.go:84: 2023-09-28 20:40:13.532 [debu]  coord1.pgcoord.querier: got agent update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a
    t.go:84: 2023-09-28 20:40:13.532 [debu]  coord2.pgcoord.querier: queried agents  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  num_agents=1  error=<nil>
    t.go:84: 2023-09-28 20:40:13.533 [debu]  coord2.pgcoord.querier: sending mappings  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  mapping_len=1
    t.go:84: 2023-09-28 20:40:13.533 [debu]  coord2.pgcoord: wrote nodes  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  nodes="[{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":2,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}]"
    t.go:84: 2023-09-28 20:40:13.533 [debu]  coord1.pgcoord.querier: queried agents  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  num_agents=1  error=<nil>
    t.go:84: 2023-09-28 20:40:13.533 [debu]  coord1.pgcoord.querier: sending mappings  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  mapping_len=1
    t.go:84: 2023-09-28 20:40:13.533 [debu]  coord1.pgcoord: got node update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  node={"id":0,"as_of":"0001-01-01T00:00:00Z","key":"nodekey:0000000000000000000000000000000000000000000000000000000000000000","disco":"discokey:0000000000000000000000000000000000000000000000000000000000000000","preferred_derp":12,"derp_latency":null,"derp_forced_websockets":null,"addresses":null,"allowed_ips":null,"endpoints":null}
    t.go:84: 2023-09-28 20:40:13.533 [debu]  coord1.pgcoord: wrote nodes  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  nodes="[{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":2,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}]"
    t.go:84: 2023-09-28 20:40:13.536 [debu]  coord2.pgcoord.querier: got client update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  client_id=e35b926b-f660-419a-8f3e-2589547896db  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a
    t.go:84: 2023-09-28 20:40:13.536 [debu]  coord1.pgcoord: upserted client subscription  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=e35b926b-f660-419a-8f3e-2589547896db  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  error=<nil>
    t.go:84: 2023-09-28 20:40:13.536 [debu]  coord1.pgcoord.querier: got client update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=e35b926b-f660-419a-8f3e-2589547896db  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a
    t.go:84: 2023-09-28 20:40:13.536 [debu]  coord1.pgcoord.querier: ignoring update because we have no mapper  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=e35b926b-f660-419a-8f3e-2589547896db  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a
    t.go:84: 2023-09-28 20:40:13.537 [debu]  coord2.pgcoord.querier: got client update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  client_id=e35b926b-f660-419a-8f3e-2589547896db  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a
    t.go:84: 2023-09-28 20:40:13.537 [debu]  coord1.pgcoord.querier: got client update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=e35b926b-f660-419a-8f3e-2589547896db  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a
    t.go:84: 2023-09-28 20:40:13.537 [debu]  coord1.pgcoord.querier: ignoring update because we have no mapper  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=e35b926b-f660-419a-8f3e-2589547896db  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a
    t.go:84: 2023-09-28 20:40:13.537 [debu]  coord1.pgcoord: upserted client binding  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=e35b926b-f660-419a-8f3e-2589547896db  node="{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":12,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}"  error=<nil>
    t.go:84: 2023-09-28 20:40:13.555 [debu]  coord2.pgcoord.querier: queried clients of agent  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  num_clients=2  error=<nil>
    t.go:84: 2023-09-28 20:40:13.555 [debu]  coord2.pgcoord.querier: sending mappings  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  mapping_len=2
    t.go:84: 2023-09-28 20:40:13.555 [debu]  coord1.pgcoord: got node update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  name=agent1  node={"id":0,"as_of":"0001-01-01T00:00:00Z","key":"nodekey:0000000000000000000000000000000000000000000000000000000000000000","disco":"discokey:0000000000000000000000000000000000000000000000000000000000000000","preferred_derp":1,"derp_latency":null,"derp_forced_websockets":null,"addresses":null,"allowed_ips":null,"endpoints":null}
    t.go:84: 2023-09-28 20:40:13.555 [debu]  coord2.pgcoord: wrote nodes  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  name=agent2  nodes="[{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":22,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null},{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":12,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}]"
    t.go:84: 2023-09-28 20:40:13.556 [debu]  coord2.pgcoord.querier: queried clients of agent  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  num_clients=2  error=<nil>
    t.go:84: 2023-09-28 20:40:13.556 [debu]  coord2.pgcoord.querier: sending mappings  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  mapping_len=2
    t.go:84: 2023-09-28 20:40:13.556 [debu]  coord2.pgcoord: wrote nodes  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  name=agent2  nodes="[{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":12,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null},{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":22,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}]"
    t.go:84: 2023-09-28 20:40:13.556 [debu]  coord1.pgcoord.querier: got agent update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339
    t.go:84: 2023-09-28 20:40:13.556 [debu]  coord1.pgcoord: upserted agent binding  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  node="{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":1,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}"  error=<nil>
    t.go:84: 2023-09-28 20:40:13.556 [debu]  coord2.pgcoord.querier: got agent update  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339
    t.go:84: 2023-09-28 20:40:13.557 [debu]  coord1.pgcoord.querier: queried agents  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  num_agents=1  error=<nil>
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord2.pgcoord.querier: queried agents  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  num_agents=1  error=<nil>
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord2.pgcoord.querier: sending mappings  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  mapping_len=1
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord1.pgcoord.querier: sending mappings  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  mapping_len=1
    t.go:84: 2023-09-28 20:40:13.558 [info]  coord2.pgcoord: closing coordinator  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord1.pgcoord: wrote nodes  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  nodes="[{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":1,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}]"
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord2.pgcoord: wrote nodes  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  nodes="[{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":1,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}]"
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord2.pgcoord: done sending updates  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord2.pgcoord: done sending updates  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord2.pgcoord: exiting recvLoop  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  error="io: read/write on closed pipe"
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord2.pgcoord: binder exiting  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  error="context canceled"
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord2.pgcoord: parent context expired while withdrawing bindings  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  error="context canceled"
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord2.pgcoord: subscriber exiting  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  error="context canceled"
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord2.pgcoord: done sending updates  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  name=agent2
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord2.pgcoord: ending heartbeats  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  error="context canceled"
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord2.pgcoord: exiting recvLoop  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  name=agent2  error="io: read/write on closed pipe"
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord2.pgcoord: ending cleanupLoop  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  error="context canceled"
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord2.pgcoord: parent context expired while withdrawing bindings  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  name=agent2  error="context canceled"
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord2.pgcoord: exiting recvLoop  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  error="io: read/write on closed pipe"
    t.go:84: 2023-09-28 20:40:13.558 [debu]  coord2.pgcoord: parent context expired while withdrawing bindings  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  error="context canceled"
    t.go:84: 2023-09-28 20:40:13.560 [debu]  coord2.pgcoord: deleted coordinator  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880
    t.go:84: 2023-09-28 20:40:13.560 [debu]  coord1.pgcoord.querier: got agent update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a
    t.go:84: 2023-09-28 20:40:13.560 [debu]  coord1.pgcoord.querier: got client update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=0ef4df4a-393a-4655-b98d-da4d1b93f6e6  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339
    t.go:84: 2023-09-28 20:40:13.560 [debu]  coord1.pgcoord.querier: got client update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=edd376b1-729a-4fdc-8efa-bb8c4fa04452  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a
    t.go:84: 2023-09-28 20:40:13.560 [debu]  coord1.pgcoord.querier: ignoring update because we have no mapper  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  client_id=edd376b1-729a-4fdc-8efa-bb8c4fa04452  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a
    t.go:84: 2023-09-28 20:40:13.561 [debu]  coord1.pgcoord.querier: queried agents  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  num_agents=0  error=<nil>
    t.go:84: 2023-09-28 20:40:13.561 [debu]  coord1.pgcoord.querier: sending mappings  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  mapping_len=0
    t.go:84: 2023-09-28 20:40:13.561 [debu]  coord1.pgcoord.querier: skipping 0 length node update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  kind=1
    t.go:84: 2023-09-28 20:40:13.561 [debu]  coord1.pgcoord.querier: queried clients of agent  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=1d7b77e4-d6ed-4f62-9e7d-b3b5c9e6f339  num_clients=1  error=<nil>
    t.go:84: 2023-09-28 20:40:13.561 [debu]  coord1.pgcoord.querier: sending mappings  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  mapping_len=1
    t.go:84: 2023-09-28 20:40:13.562 [debu]  coord1.pgcoord: wrote nodes  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  name=agent1  nodes="[{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":11,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}]"
    t.go:84: 2023-09-28 20:40:15.503 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:15.503 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:17.494 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:17.494 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:19.494 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:19.494 [debu]  coord2.pgcoord: checking heartbeat expiry  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880
    t.go:84: 2023-09-28 20:40:19.494 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:19.494 [debu]  coord2.pgcoord: last heartbeat from coordinator  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  other_coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  last_heartbeat=6.000128539s
    t.go:84: 2023-09-28 20:40:19.494 [info]  coord2.pgcoord: coordinator failed heartbeat check  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  other_coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  last_heartbeat=6.000128539s
    t.go:84: 2023-09-28 20:40:19.494 [debu]  coord2.pgcoord: computed oldest heartbeat  coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  oldest="0001-01-01T00:00:00Z"  time_to_expiry=-2562047h47m16.854775808s
    t.go:84: 2023-09-28 20:40:19.505 [debu]  coord1.pgcoord: checking heartbeat expiry  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:19.505 [debu]  coord1.pgcoord: last heartbeat from coordinator  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  other_coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  last_heartbeat=6.00068[360](https://github.com/coder/coder/actions/runs/6344009627/job/17233161856?pr=9920#step:5:361)7s
    t.go:84: 2023-09-28 20:40:19.505 [info]  coord1.pgcoord: coordinator failed heartbeat check  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  other_coordinator_id=3435eb50-dce9-4c0b-ad7e-806f9030a880  last_heartbeat=6.000683607s
    t.go:84: 2023-09-28 20:40:19.505 [debu]  coord1.pgcoord: computed oldest heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  oldest="0001-01-01T00:00:00Z"  time_to_expiry=-2562047h47m16.854775808s
    t.go:84: 2023-09-28 20:40:19.505 [debu]  coord1.pgcoord: skipping duplicate update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  nodes="[{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":1,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}]"
    t.go:84: 2023-09-28 20:40:19.505 [debu]  coord1.pgcoord: skipping duplicate update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  name=agent1  nodes="[{\"id\":0,\"as_of\":\"0001-01-01T00:00:00Z\",\"key\":\"nodekey:0000000000000000000000000000000000000000000000000000000000000000\",\"disco\":\"discokey:0000000000000000000000000000000000000000000000000000000000000000\",\"preferred_derp\":11,\"derp_latency\":null,\"derp_forced_websockets\":null,\"addresses\":null,\"allowed_ips\":null,\"endpoints\":null}]"
    t.go:84: 2023-09-28 20:40:19.505 [debu]  coord1.pgcoord.querier: skipping 0 length node update  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  agent_id=971e1858-6b12-4913-ac5e-b9043a738d5a  kind=1
    t.go:84: 2023-09-28 20:40:21.493 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:21.493 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:23.492 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:23.492 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:25.492 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:25.492 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:27.493 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:27.493 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:29.494 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:29.494 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:31.492 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:31.492 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:33.493 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:33.493 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:35.494 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:35.494 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:37.494 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:37.494 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:39.493 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:39.493 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:41.494 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:41.494 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:43.493 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:43.493 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:45.493 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:45.493 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:47.494 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:47.494 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:49.493 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:49.493 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:51.494 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:51.494 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:53.493 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:53.493 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
make: *** [Makefile:672: test-postgres] Error 1
    t.go:84: 2023-09-28 20:40:55.493 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:55.493 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:57.494 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:57.494 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:59.493 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:40:59.493 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:41:01.494 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:41:01.494 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:41:03.493 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:41:03.493 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:41:05.494 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:41:05.494 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:41:07.494 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:41:07.494 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:41:09.493 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:41:09.493 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:41:11.494 [debu]  coord1.pgcoord: sent heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:41:11.494 [debu]  coord1.pgcoord: ignoring our own heartbeat  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    pgcoord_test.go:[382](https://github.com/coder/coder/actions/runs/6344009627/job/17233161856?pr=9920#step:5:383): timeout receiving error
    t.go:84: 2023-09-28 20:41:13.479 [debu]  coord1.pgcoord: done sending updates  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:41:13.479 [debu]  coord1.pgcoord: exiting recvLoop  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  name=agent1  error="io: read/write on closed pipe"
    t.go:84: 2023-09-28 20:41:13.479 [info]  coord2.pgcoord: closing coordinator  coordinator_id=3[435](https://github.com/coder/coder/actions/runs/6344009627/job/17233161856?pr=9920#step:5:436)eb50-dce9-4c0b-ad7e-806f9030a880
    t.go:84: 2023-09-28 20:41:13.[479](https://github.com/coder/coder/actions/runs/6344009627/job/17233161856?pr=9920#step:5:480) [info]  coord1.pgcoord: closing coordinator  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:41:13.479 [debu]  coord1.pgcoord: ending heartbeats  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  error={}
    t.go:84: 2023-09-28 20:41:13.[480](https://github.com/coder/coder/actions/runs/6344009627/job/17233161856?pr=9920#step:5:481) [debu]  coord1.pgcoord: subscriber exiting  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  error={}
    t.go:84: 2023-09-28 20:41:13.480 [debu]  coord1.pgcoord: exiting recvLoop  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  error=EOF
    t.go:84: 2023-09-28 20:41:13.480 [debu]  coord1.pgcoord: binder exiting  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  error={}
    t.go:84: 2023-09-28 20:41:13.480 [debu]  coord1.pgcoord: ending cleanupLoop  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  error={}
    t.go:84: 2023-09-28 20:41:13.479 [debu]  coord1.pgcoord: done sending updates  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9
    t.go:84: 2023-09-28 20:41:13.479 [debu]  coord1.pgcoord: exiting recvLoop  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab[532](https://github.com/coder/coder/actions/runs/6344009627/job/17233161856?pr=9920#step:5:533)d9  error=EOF
    t.go:84: 2023-09-28 20:41:13.479 [debu]  coord1.pgcoord: done sending updates  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  name=agent1
    t.go:84: 2023-09-28 20:41:13.479 [debu]  coord1.pgcoord: parent context expired while withdrawing bindings  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  name=agent1  error={}
    t.go:84: 2023-09-28 20:41:13.480 [debu]  coord1.pgcoord: parent context expired while withdrawing bindings  coordinator_id=e0e12d48-5b95-4dcc-baa5-77a7dab532d9  error={}
```
</details>

--- a/enterprise/tailnet/pgcoord_test.go
+++ b/enterprise/tailnet/pgcoord_test.go
@@ -645,12 +645,20 @@ func (c *testConn) recvNodes(ctx context.Context, t *testing.T) []*agpl.Node {
 
 func (c *testConn) recvErr(ctx context.Context, t *testing.T) error {
 	t.Helper()
-	select {
-	case <-ctx.Done():
-		t.Fatal("timeout receiving error")
-		return ctx.Err()
-	case err := <-c.errChan:
-		return err
+	// pgCoord works on eventual consistency, so it sometimes sends extra node
+	// updates, and these block errors if not read from the nodes channel.
+	for {
+		select {
+		case nodes := <-c.nodeChan:
+			t.Logf("ignoring nodes update while waiting for error; id=%s, nodes=%+v",
+				c.id.String(), nodes)
+			continue
+		case <-ctx.Done():
+			t.Fatal("timeout receiving error")
+			return ctx.Err()
+		case err := <-c.errChan:
+			return err
+		}
 	}
 }
 

